rules/manager.go

```diff
@@ -115,6 +117,22 @@ func NewGroupMetrics(reg prometheus.Registerer) *Metrics {
 			},
 			[]string{"rule_group"},
 		),
+		ingestionSamplesTotal: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: namespace,
+				Name:      "rule_result_ingestion_samples_total",
+				Help:      "The total number of result samples generated for ingestion.",
+			},
+			[]string{"rule_group"},
+		),
+		ingestionFailuresTotal: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: namespace,
+				Name:      "rule_result_ingestion_sample_failures_total",
```

Comment:
There should be a metric per failure type, to match scraping.

Reply:
I have tried to follow scraping as much as possible, but there needs to be a catch all for errors that don't match the other categories which I called `other_failures`